lib/logstash/outputs/logstash.rb

```diff
@@ -61,129 +54,190 @@ def initialize(*a)
       fail LogStash::ConfigurationError, 'The `logstash` output does not have an externally-configurable `codec`'
     end
 
-    if @ssl_certificate_authorities && @ssl_certificate_authorities.size > 1
-      fail LogStash::ConfigurationError, 'The `logstash` output supports at most one `ssl_certificate_authorities` path'
+    @headers = {}.tap do | header |
+      header["Content-Type"] = "application/x-ndjson"
+      header["Content-Encoding"] = "gzip"
     end
 
-    logger.debug("initializing inner HTTP output plugin")
-    @internal_http = plugin_factory.output('http').new(inner_http_output_options)
-    logger.debug("inner HTTP output plugin has been initialized")
+    logger.debug("`logstash` output plugin has been initialized.")
   end
 
   def register
-    logger.debug("registering inner HTTP output plugin")
-    @internal_http.register
-    logger.debug("inner HTTP output plugin has been registered")
+    logger.debug("Registering `logstash` output plugin.")
+
+    # TODO: confirm if we have a plan to change @user to @username
+    logger.warn("Both `user` and `username` are defined, using `username` value.") if @user && @username
+    @user = @username ? @username.freeze : @user
```

Comment:
I would prefer deprecating one and standardizing on the other (I personally prefer `username`). We could use the Config Normalization Support mixin to handle this.

Reply:
Hm, it seems we cannot use normalization support unless we deprecate the `user`. I have tried (wonder if I missed anything) and `with_deprecated_alias` method (through `with_deprecated_mapping` -> `ensure_deprecated!`) of the normalization support mixin package complains `Config `user` not marked deprecated`.

```
require 'logstash/plugin_mixins/normalize_config_support'
...
include LogStash::PluginMixins::NormalizeConfigSupport
...
@username = normalize_config(:username) do |normalize|
  normalize.with_deprecated_alias(:user)
end
```

Reply:
Applied the `user` deprecation with [recent commit](https://github.com/logstash-plugins/logstash-integration-logstash/pull/16/commits/9ee706c755232d80ae251b8507af2394a2010a96).